universal-application-tool-0.0.1/conf/routes

```diff
@@ -14,6 +14,7 @@ GET     /postgres                   controllers.PostgresController.list()
 
 # Add a person
 GET     /postgres/add/:name         controllers.PostgresController.add(name: String)
+GET     /postgres/sync_add/:name    controllers.PostgresController.syncAdd(name: String)
```

Comment:
This and the other `add` method should both be POST requests. The data for the new record should be provided in the request body rather than the URL.

https://www.restapitutorial.com/lessons/httpmethods.html

Reply:
That makes sense and we should totally do it. I incline to leave it for a later change, e.g. when we add forms and logic for form binding, if that is okay with you.

Reply:
The intention of this code is to serve as an example to others, so I'd like for this and the other create method to be corrected. I'm a little concerned about about approving code that uses the current.

As far as the scope of the change goes, I think the steps are:

Update the route(s):
```
POST     /postgres   controllers.PostgresController.create(request: Request)
POST     /postgres/sync  controllers.PostgresController.createSync(request: Request)
```

Update the test to use post and specify a form body for the request:
```java
fakeRequest(routes.PostgresController.createSync())
            .method(play.test.Helpers.POST)
            .header(Http.HeaderNames.HOST, "localhost:" + testServerPort())
            .bodyForm(ImmutableMap.of("name", "John"));
```

Get the name out of the request body:
```java
  public Result createSync(Request request) {
    DynamicForm requestData = formFactory.form().bindFromRequest(request);
    Person person = new Person();
    person.name = requestData.get("name");
    ...
  }
```

Reply:
Done. Please take a look.